package dmGamesysDDF;

import "ddf/ddf_extensions.proto";
import "ddf/ddf_math.proto";

option java_package = "com.dynamo.camera.proto";
option java_outer_classname = "Camera";

message CameraDesc
{
    required float  aspect_ratio        = 1;
    required float  fov                 = 2;
    required float  near_z              = 3;
    required float  far_z               = 4;
    optional uint32 auto_aspect_ratio   = 5 [default = 0];
}

/*#
 * Set camera message
 * @message
 * @name SetCamera
 * @param aspect_ratio aspect_ratio to set
 * @param fov field of view in radians to set
 * @param near_z near z plane to set
 * @param far_z far z plane to set
 */
message SetCamera
{
    required float          aspect_ratio    = 1;
    required float          fov             = 2;
    required float          near_z          = 3;
    required float          far_z           = 4;
}

message AddCameraTarget
{
    required dmMath.Point3 aabb_min = 1;
    required dmMath.Point3 aabb_max = 2;
    required uint32 high_prio       = 3;
}
